TypeCobol.Test/Utils/Comparator.cs

```diff
@@ -1136,6 +1110,150 @@ private void WriteTypeDefChildrens(StringBuilder sb, DocumentationTypeChildren c
+        }
+
+        public ChangeComparator(Paths path, bool debug = false, bool isEI = false)
+            : base(path, debug, isEI)
+        {
+
+        }
+
+        public override void Compare(CompilationUnit result, IncrementalChangesHistory history, StreamReader reader, string expectedResultPath)
+        {
+            // Dump full incremental history
+            var actual = new StringBuilder();
+            DumpEvents(history.TextChangedEvents, DumpCobolTextLine);
+            DumpEvents(history.TokensChangedEvents, DumpTokensLine);
+            DumpEvents(history.ProcessedTokensChangedEvents, DumpProcessedTokensLine);
+            DumpEvents(history.CodeElementsChangedEvents, DumpCodeElementsLine);
+
+            // Compare with expected
+            ParserUtils.CheckWithResultReader(paths.SamplePath, actual.ToString(), reader, expectedResultPath);
+
+            void WriteSeparator(string title, char separatorChar = '=')
+            {
+                int paddingLength = 78 - title.Length;
+                string left = new string(separatorChar, paddingLength / 2);
+                string right = new string(separatorChar, paddingLength / 2 + paddingLength % 2);
+                actual.AppendLine($"{left} {title} {right}");
+            }
+
+            void DumpEvents<TLine>(IEnumerable<DocumentChangedEvent<TLine>> changeEvents, Action<StringBuilder, TLine> dumpLine)
+                where TLine : ICobolTextLine
+            {
+                WriteSeparator(typeof(TLine).Name);
+
+                bool hasChanges = false;
+                DocumentVersion<TLine> first = null;
+                DocumentVersion<TLine> last = null;
+                foreach (var changeEvent in changeEvents)
+                {
+                    if (first == null)
+                    {
+                        first = changeEvent.DocumentVersionBefore;
+                    }
+
+                    last = changeEvent.DocumentVersionAfter;
+
+                    DumpChanges(changeEvent.DocumentChanges);
+                }
+
+                if (hasChanges)
+                {
+                    if (first != null && last != null && first != last)
+                    {
+                        WriteSeparator(nameof(DocumentVersion<TLine>.GetReducedAndOrderedChangesInNewerVersion), '-');
+                        var reducedChanges = first.GetReducedAndOrderedChangesInNewerVersion(last);
+                        DumpChanges(reducedChanges);
+                    }
+                }
+                else
+                {
+                    actual.AppendLine("No change");
+                }
+
+                void DumpChanges(IEnumerable<DocumentChange<TLine>> documentChanges)
+                {
+                    foreach (var documentChange in documentChanges)
+                    {
+                        hasChanges = true;
+                        actual.Append($"Line {documentChange.LineIndex}: {documentChange.Type} -> ");
```

Comment:
Can you align the `:` after the LineIndex when LineIndex have different length ?
Can you align the arrow `->` for every `documentChange.Type` ?
The idea is to make it easy to compare different lines:
```
Line 0: LineInserted -> [8,20:             ]<SpaceSeparator> 
Line 0: LineUpdated -> [8,20:             ]<SpaceSeparator> 
Line 99: LineInserted -> [8,20:             ]<SpaceSeparator> 
Line 100: LineUpdated -> [8,20:             ]<SpaceSeparator> 
```
Should be:
```
Line   0: LineInserted -> [8,20:             ]<SpaceSeparator> 
Line   0: LineUpdated  -> [8,20:             ]<SpaceSeparator> 
Line  99: LineInserted -> [8,20:             ]<SpaceSeparator> 
Line 100: LineUpdated  -> [8,20:             ]<SpaceSeparator> 
```

Reply:
Yes. See updated output.